cmd/antrea-agent/options.go

```diff
@@ -207,69 +151,16 @@ func (o *Options) setDefaults() {
 	if o.config.OVSRunDir == "" {
 		o.config.OVSRunDir = ovsconfig.DefaultOVSRunDir
 	}
-	if o.config.HostGateway == "" {
-		o.config.HostGateway = defaultHostGateway
-	}
-	if o.config.TrafficEncapMode == "" {
-		o.config.TrafficEncapMode = config.TrafficEncapModeEncap.String()
-	}
-	if o.config.TrafficEncryptionMode == "" {
-		o.config.TrafficEncryptionMode = config.TrafficEncryptionModeNone.String()
-	}
-	if o.config.TunnelType == "" {
-		o.config.TunnelType = defaultTunnelType
-	}
-	if o.config.HostProcPathPrefix == "" {
-		o.config.HostProcPathPrefix = defaultHostProcPathPrefix
-	}
-	if features.DefaultFeatureGate.Enabled(features.AntreaProxy) {
-		if o.config.AntreaProxy.ProxyLoadBalancerIPs == nil {
-			o.config.AntreaProxy.ProxyLoadBalancerIPs = new(bool)
-			*o.config.AntreaProxy.ProxyLoadBalancerIPs = true
-		}
-	} else {
-		if o.config.ServiceCIDR == "" {
-			o.config.ServiceCIDR = defaultServiceCIDR
-		}
-	}
 	if o.config.APIPort == 0 {
 		o.config.APIPort = apis.AntreaAgentAPIPort
 	}
-	if o.config.ClusterMembershipPort == 0 {
-		o.config.ClusterMembershipPort = apis.AntreaAgentClusterMembershipPort
-	}
-	if o.config.EnablePrometheusMetrics == nil {
-		o.config.EnablePrometheusMetrics = new(bool)
-		*o.config.EnablePrometheusMetrics = true
+	if o.config.NodeType == "" {
+		o.config.NodeType = defaultNodeType.String()
 	}
-	if o.config.WireGuard.Port == 0 {
-		o.config.WireGuard.Port = apis.WireGuardListenPort
-	}
-
-	if features.DefaultFeatureGate.Enabled(features.FlowExporter) {
-		if o.config.FlowCollectorAddr == "" {
-			o.config.FlowCollectorAddr = defaultFlowCollectorAddress
-		}
-		if o.config.FlowPollInterval == "" {
-			o.pollInterval = defaultFlowPollInterval
-		}
-		if o.config.ActiveFlowExportTimeout == "" {
-			o.activeFlowTimeout = defaultActiveFlowExportTimeout
-		}
-		if o.config.IdleFlowExportTimeout == "" {
-			o.idleFlowTimeout = defaultIdleFlowExportTimeout
-		}
-	}
-
-	if features.DefaultFeatureGate.Enabled(features.NodePortLocal) {
-		switch {
-		case o.config.NodePortLocal.PortRange != "":
-		case o.config.NPLPortRange != "":
-			klog.InfoS("The nplPortRange option is deprecated, please use nodePortLocal.portRange instead")
-			o.config.NodePortLocal.PortRange = o.config.NPLPortRange
-		default:
-			o.config.NodePortLocal.PortRange = defaultNPLPortRange
-		}
+	if strings.EqualFold(o.config.NodeType, config.K8sNode.String()) {
```

Comment:
Supporting only exact case would make code simpler. We never heard people complain values like `Geneve`, `GENEVE`, `Netdev` are not accepted. I know trafficEncapMode is an exception, I'm thinking unifying it in case it will be taken as example for more new code, and cause inconsistent user experience that some config accepts both cases while some don't.